committeeoversightapp/views.py

```diff
@@ -1,3 +1,160 @@
+        committee_form = CommitteeForm(request.POST, prefix="committee")
+        committeemember_form = CommitteeMemberForm(request.POST, prefix="committeemember")
+        witness_form = WitnessForm(request.POST, prefix="witness")
+        transcript_form = TranscriptForm(request.POST, prefix="transcript")
+
+        print("Checking if forms are valid...")
+
+        forms_valid = [event_form.is_valid(), committee_form.is_valid(),
+                       committeemember_form.is_valid(), witness_form.is_valid(),
+                       transcript_form.is_valid()]
+
+        if all(forms_valid):
+            print("All forms valid! Saving...")
+
+            # save event
+            event = event_form.save()
+
+            # find and create committees as EventParticipants
+            committees = committee_form.cleaned_data['name']
+
+            for committee in committees:
+                name = committee.name
+                event = event_form.save(commit=False)
+                organization = Organization.objects.get(id=committee.id)
+                entity_type = "organization"
+                new_committee = EventParticipant(name=name, event=event, organization=organization, entity_type=entity_type)
+                new_committee.save()
+
+            # find and create committee members as EventParticipants
+            committeemembers = committeemember_form.cleaned_data['name'].split(",")
+
+            for committeemember in committeemembers:
+                if committeemember == '' or committeemember.isspace():
+                    pass
+                else:
+                    name = committeemember.strip()
+                    entity_type = "committee member"
+                    new_committeemember = EventParticipant(name=name, event=event, entity_type=entity_type)
+                    new_committeemember.save()
+
+            # find and create witnesses as EventParticipants
+            witnesses = witness_form.cleaned_data['name'].split(",")
+
+            for witness in witnesses:
+                if witness == '' or witness.isspace():
+                    pass
+                else:
+                    name = witness.strip()
+                    entity_type = "person"
+                    new_witness = EventParticipant(name=name, event=event, entity_type=entity_type)
+                    new_witness.save()
+
+            # if form includes a transcript URL create EventDocument with original and archived url
+            transcript_url = transcript_form.cleaned_data['url']
+
+            if transcript_url == '' or witness.isspace():
+                pass
+            else:
+                note="transcript"
+                new_document = EventDocument(note=note, event=event)
+                new_document.save()
+
+                archived_transcript_url = archive_url(transcript_url)
+                print(archived_transcript_url)
+
+                ext = get_ext(transcript_url)
+                if ext.lower() == '.pdf':
```

Comment:
You could make this a tad more DRY. One way that comes to mind: a mapping of the extensions, rather than the successive IF/ELSE statements. So, something like: 

```
extensions = {'.pdf': 'application/pdf', '.htm': 'text/html'} 
...
media_type=extensions[ext.lower()]
```

Reply:
oo clever!